runtime/compiler/env/j9method.cpp

```diff
@@ -2387,15 +2393,48 @@ TR_ResolvedJ9Method::TR_ResolvedJ9Method(TR_FrontEnd * fe, TR_ResolvedMethod * o
 
 void TR_ResolvedJ9Method::construct()
    {
-#define x(a, b, c) a, sizeof(b) - 1, b, (int16_t)strlen(c), c
-
    struct X
       {
       TR::RecognizedMethod _enum;
       char _nameLen;
       const char * _name;
       int16_t _sigLen;
       const char * _sig;
+
+      typedef int8_t javaVersion_t;
+
+      javaVersion_t _nativeMinVersion;
+      javaVersion_t _nativeMaxVersion;
+
+// AllPastJavaVer and AllFutureJavaVer are used as lower and upper bounds for
+// ranges of Java versions over which a recognized method is expected to be
+// native, where that range is (currently) unbounded.  An upper bound of
+// AllPastJaveVer is used if the method has never been expected to be native.
+#define AllPastJavaVer   INT8_MIN
+#define AllFutureJavaVer INT8_MAX
+
+// xClMeth, xSig, xWCSig and xNonNativeRange are intermediate macros used in
+// the macros used to initialize entries in the tables of recognized methods
+#define xClMeth(cl, meth) cl, sizeof(meth) - 1, meth
+#define xSig(sig)         (int16_t) sizeof(sig) - 1, sig
+#define xWCSig            (int16_t) -1, "*"
+#define xNonNativeRange   AllPastJavaVer, AllPastJavaVer
```

Comment:
This seems odd - why isn't it AllPastJavaVer to AllFutuerJaveVer by default?

Reply:
The version ranges are supposed to indicate for which versions a recognized method is expected to be native.  `xNonNativeRange` is intended to be used with recognized methods that have never been expected to be native.

Would renaming it to something like `xNeverNativeRange` or perhaps `xNeverBeenNativeRange` help clarify?

Similarly, would changing the name of `AllPastJavaVer` to something like `InfinitePast` or `BoundlessPast` and `AllFutureJavaVer` to something like `InfiniteFuture` or `BoundlessFuture` help clarify their purposes.  Then the definition of this macro would become:

```
#define xNeverBeenNativeRange   InfinitePast, InfinitePast
```

to suggest that this method was native in the range [-∞, -∞].  ([-infinity, -infinity] if your browser can't render that.)

Other suggestions will be gladly entertained!